src/main/java/com/direwolf20/buildinggadgets/client/gui/materiallist/ScrollingMaterialList.java

```diff
@@ -1,214 +1,224 @@
-//                int lineXEnd = right - widthAmount - LINE_SIDE_MARGIN;
-//                int lineY = AlignmentUtil.getYForAlignedCenter(1, top, bottom - 1);
-//                GlStateManager.enableAlphaTest();
-//                GlStateManager.enableBlend();
-//                GlStateManager.blendFunc(GL_SRC_ALPHA, GL_ONE_MINUS_SRC_ALPHA);
-//                parent.hLine(lineXStart, lineXEnd, lineY, 0x22FFFFFF);
+package com.direwolf20.buildinggadgets.client.gui.materiallist;
+
+import com.direwolf20.buildinggadgets.api.building.view.*;
+import com.direwolf20.buildinggadgets.api.materials.UniqueItem;
+import com.direwolf20.buildinggadgets.api.template.IBuildOpenOptions;
+import com.direwolf20.buildinggadgets.api.template.SimpleBuildOpenOptions;
+import com.direwolf20.buildinggadgets.client.gui.base.EntryList;
+import com.direwolf20.buildinggadgets.common.util.helpers.InventoryHelper;
+import com.google.common.collect.Multiset;
+import com.mojang.blaze3d.platform.GlStateManager;
+import net.minecraft.client.Minecraft;
+import net.minecraft.client.entity.player.ClientPlayerEntity;
+import net.minecraft.client.gui.widget.list.ExtendedList;
+import net.minecraft.client.renderer.RenderHelper;
+import net.minecraft.item.ItemStack;
+import net.minecraft.util.math.MathHelper;
+import net.minecraft.world.World;
+import org.lwjgl.glfw.GLFW;
+
+import java.awt.*;
+import java.util.Objects;
+
+import static com.direwolf20.buildinggadgets.client.gui.materiallist.MaterialListGUI.*;
+import static com.direwolf20.buildinggadgets.client.gui.materiallist.ScrollingMaterialList.Entry;
+import static org.lwjgl.opengl.GL11.*;
+
+class ScrollingMaterialList extends EntryList<Entry> {
+
+    static final int TOP = 16;
+    static final int BOTTOM = 32;
+
+    private static final int SLOT_SIZE = 18;
+    private static final int MARGIN = 2;
+    private static final int ENTRY_HEIGHT = Math.max(SLOT_SIZE + MARGIN * 2, Minecraft.getInstance().fontRenderer.FONT_HEIGHT * 2 + MARGIN * 3);
+    private static final int LINE_SIDE_MARGIN = 8;
+
+    private MaterialListGUI gui;
+
+    private SortingModes sortingMode;
+
+    public ScrollingMaterialList(MaterialListGUI gui) {
+        super(gui.getWindowLeftX(),
+                gui.getWindowTopY() + TOP,
+                gui.getWindowWidth(),
+                gui.getWindowHeight() - TOP - BOTTOM,
+                ENTRY_HEIGHT);
+        this.gui = gui;
+
+        World world = Minecraft.getInstance().world;
+        ClientPlayerEntity player = Minecraft.getInstance().player;
+        IBuildContext context = SimpleBuildContext.builder()
+                .buildingPlayer(player)
+                .usedStack(gui.getTemplateItem())
+                .build(world);
+        IBuildOpenOptions options = SimpleBuildOpenOptions.withContext(context);
+        IBuildView view = Objects.requireNonNull(gui.getTemplateCapability().createViewInContext(options));
```

Comment:
Ouch, I missed that on the first pass... You forgot to close the build view! Notice that IBuildView implements AutoClosable for a reason: it blocks Transactions. Theredore you hereby prevent any Transactions from occuring after this has been run.

Reply:
Also, I'm unsure whether this is the correft way to go... You might need to update the List, if the Template get's synced from the Server - aka I think you need a listener and in that either recreate the Scrolling MaterialList or re-execute this code.